### ISSUE TITLE
[google_sign_in] [iOS] Can't sign in anymore after closing the signin webview

### ISSUE THREAD

**Issue by fromlabs** [flutter/flutter#84424]
Hi,

I've noticed that using an iPad, I can close the signin webview tapping outside. This action doesn't properly cancel the signin flow and it's not possible to signin anymore (only after an app restart).

`google_sign_in: ^5.0.4`

```
[✓] Flutter (Channel unknown, 2.0.6, on macOS 11.4 20F71 darwin-x64, locale it-IT)
    • Flutter version 2.0.6 at /Users/hans/fvm/versions/2.0.6
    • Framework revision 1d9032c7e1 (6 weeks ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/hans/Library/Android/sdk
    • Platform android-29, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (2 available)
    • iPad di Roberto Tassi (mobile) • 6a40d226a7e24d8b592c21cd8956266daa9efcbf • ios            • iOS 14.4.2
    • Chrome (web)                   • chrome                                   • web-javascript • Google Chrome 91.0.4472.101

• No issues found!
```

**Comment by TahaTesser** [comment]
Hi @fromlabs 
Can you please provide your `flutter doctor -v`, a complete reproducible minimal code sample, and complete `flutter run --verbose` logs running the minimal code sample?
Thank you

**Comment by darshankawar** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
